sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -307,6 +308,8 @@ class JacksonParser(
     }
   }
 
+  private lazy val allowEmptyString = SQLConf.get.getConf(SQLConf.LEGACY_ALLOW_EMPTY_STRING_IN_JSON)
```

Comment:
We could avoid the overhead of checking **lazy** val. Do you see any problem if it will be just val?

Reply:
Just think it is not used all the time. Let me make it as val as it is cheap.

Reply:
Yeh, but lazy val consumes memory as well even if it is not touched. Need to maintain an internal flag to track that the `val` was touched or not.